### DISCUSSION TITLE
Error occurred prerendering page "/500" [Any experience build/deploying with Lerna monorepo and Next project]

### DISCUSSION THREAD

**Discussion by icbrahimc** [vercel/next.js#66766] [Help]
### Summary

I'm experiencing an issue while deploying my Next.js project via AWS Amplify. My Next.js project is part of a Lerna monorepo. When I attempt to run npm run build locally or deploy via AWS Amplify, I encounter the following error:

```
   Generating static pages (0/10)  [=   ]Error: Minified React error #321; visit https://reactjs.org/docs/error-decoder.html?invariant=321 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at S (/Users/ibrahimconteh/work/groovy/node_modules/react/cjs/react.production.min.js:18:327)
    at exports.useContext (/Users/ibrahimconteh/work/groovy/node_modules/react/cjs/react.production.min.js:22:269)
    at StyleRegistry (/Users/ibrahimconteh/work/groovy/node_modules/styled-jsx/dist/index/index.js:450:30)
    at Wc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
    at Zc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
    at Z (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at Zc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:74:209)
    at Z (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at Zc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:74:209)
    at Z (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)

Error occurred prerendering page "/500". Read more: https://nextjs.org/docs/messages/prerender-error

Error: Minified React error #321; visit https://reactjs.org/docs/error-decoder.html?invariant=321 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at S (/Users/ibrahimconteh/work/groovy/node_modules/react/cjs/react.production.min.js:18:327)
    at exports.useContext (/Users/ibrahimconteh/work/groovy/node_modules/react/cjs/react.production.min.js:22:269)
    at StyleRegistry (/Users/ibrahimconteh/work/groovy/node_modules/styled-jsx/dist/index/index.js:450:30)
    at Wc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
    at Zc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
    at Z (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at Zc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:74:209)
    at Z (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at Zc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:74:209)
    at Z (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
Error: Minified React error #321; visit https://reactjs.org/docs/error-decoder.html?invariant=321 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at S (/Users/ibrahimconteh/work/groovy/node_modules/react/cjs/react.production.min.js:18:327)
    at exports.useContext (/Users/ibrahimconteh/work/groovy/node_modules/react/cjs/react.production.min.js:22:269)
    at StyleRegistry (/Users/ibrahimconteh/work/groovy/node_modules/styled-jsx/dist/index/index.js:450:30)
    at Wc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
    at Zc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
    at Z (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at Zc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:74:209)
    at Z (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at Zc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:74:209)
    at Z (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)

Error occurred prerendering page "/404". Read more: https://nextjs.org/docs/messages/prerender-error

Error: Minified React error #321; visit https://reactjs.org/docs/error-decoder.html?invariant=321 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at S (/Users/ibrahimconteh/work/groovy/node_modules/react/cjs/react.production.min.js:18:327)
    at exports.useContext (/Users/ibrahimconteh/work/groovy/node_modules/react/cjs/react.production.min.js:22:269)
    at StyleRegistry (/Users/ibrahimconteh/work/groovy/node_modules/styled-jsx/dist/index/index.js:450:30)
    at Wc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
    at Zc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
    at Z (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at Zc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:74:209)
    at Z (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at Zc (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:74:209)
    at Z (/Users/ibrahimconteh/work/groovy/packages/admin-portal/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
 ✓ Generating static pages (10/10)

> Export encountered errors on following paths:
        /_error: /404
        /_error: /500
```

The error occurs during the prerendering of static pages such as /404 and /500. The error traceback points to issues with react, react-dom, and styled-jsx during the server-side rendering process.

Interestingly, the project builds correctly when run outside of the monorepo setup. This suggests that the issue might be related to the way dependencies are managed within the monorepo.

Here's my package.json

```
{
  "license": "MIT",
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "lint:fix": "eslint --ext .ts --fix --max-warnings 10 ./app"
  },
  "dependencies": {
    "@headlessui/react": "^1.7.17",
    "@heroicons/react": "^2.0.18",
    "@material-tailwind/html": "^2.2.2",
    "@paralleldrive/cuid2": "^2.2.2",
    "@remixicon/react": "^4.2.0",
    "@tanstack/react-table": "^8.17.0",
    "@tremor/react": "^3.10.0",
    "@types/js-cookie": "^3.0.5",
    "autoprefixer": "^10.4.16",
    "classnames": "^2.5.1",
    "dayjs": "^1.11.10",
    "formik": "^2.4.5",
    "js-cookie": "^3.0.5",
    "next": "14.2.3",
    "next-auth": "^5.0.0-beta.19",
    "prettier": "^3.0.3",
    "prop-types": "^15.8.1",
    "react": "^18",
    "react-dom": "^18",
    "react-hot-toast": "^2.4.1",
    "react-select": "^5.8.0",
    "sass": "^1.74.1",
    "server-only": "^0.0.1",
    "yup": "^1.4.0"
  },
  "prettier": {
    "arrowParens": "always",
    "singleQuote": true,
    "tabWidth": 2,
    "trailingComma": "none"
  },
  "devDependencies": {
    "@babel/eslint-parser": "^7.24.7",
    "@babel/preset-react": "^7.24.7",
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "eslint": "^8",
    "eslint-config-next": "^14.2.3",
    "eslint-plugin-react-hooks": "^4.6.2",
    "postcss": "^8",
    "tailwindcss": "^3.4.1",
    "typescript": "^5"
  }
}
```

If anyone has any suggestions please advise! Been wasting a few days on trying to get this to work

**Comment by davecarlson** [comment]
I bumped react and react-dom from 18.2 to 18.3.1 and now consistantly get the same issue. Downgraded to 18.2, error went away.

try pinning to 18.2 and see if it fixes it for you? I have no idea of course the underlying issue

**Comment by mmopinodo** [comment]
i have a similar problem, using these versions: 
```
    "next": "15.0.0-rc.0",
    "optional": "^0.1.4",
    "next-logger": "^4.0.0",
    "otplib": "^12.0.1",
    "papaparse": "^5.4.1",
    "posthog-js": "^1.135.2",
    "prismjs": "^1.29.0",
    "qrcode": "^1.5.3",
    "react": "19.0.0-rc-107a2f8c3e-20240617",
    "react-dom": "19.0.0-rc-107a2f8c3e-20240617",
```

```
   Generating static pages (0/57)  [ ===]TypeError: Cannot read properties of null (reading 'useContext')
@formbricks/web:build:     at exports.useContext (/Users/xstraza/OpinodoProjects/opinodo-surveys/apps/web/node_modules/react/cjs/react.production.js:489:33)
@formbricks/web:build:     at h (/Users/xstraza/OpinodoProjects/opinodo-surveys/apps/web/.next/server/pages/_error.js:1:6021)
@formbricks/web:build:     at renderWithHooks (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4007:18)
@formbricks/web:build:     at renderElement (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4166:17)
@formbricks/web:build:     at renderNodeDestructive (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4696:16)
@formbricks/web:build:     at renderNode (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4921:14)
@formbricks/web:build:     at renderChildrenArray (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4853:7)
@formbricks/web:build:     at renderNodeDestructive (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4710:9)
@formbricks/web:build:     at renderNode (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4921:14)
@formbricks/web:build:     at renderElement (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4207:7)
@formbricks/web:build: 
@formbricks/web:build: Error occurred prerendering page "/500". Read more: https://nextjs.org/docs/messages/prerender-error
@formbricks/web:build: 
@formbricks/web:build: TypeError: Cannot read properties of null (reading 'useContext')
@formbricks/web:build:     at exports.useContext (/Users/xstraza/OpinodoProjects/opinodo-surveys/apps/web/node_modules/react/cjs/react.production.js:489:33)
@formbricks/web:build:     at h (/Users/xstraza/OpinodoProjects/opinodo-surveys/apps/web/.next/server/pages/_error.js:1:6021)
@formbricks/web:build:     at renderWithHooks (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4007:18)
@formbricks/web:build:     at renderElement (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4166:17)
@formbricks/web:build:     at renderNodeDestructive (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4696:16)
@formbricks/web:build:     at renderNode (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4921:14)
@formbricks/web:build:     at renderChildrenArray (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4853:7)
@formbricks/web:build:     at renderNodeDestructive (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4710:9)
@formbricks/web:build:     at renderNode (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4921:14)
@formbricks/web:build:     at renderElement (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4207:7)

> Build error occurred
Error: Export encountered an error on /500, exiting due to prerenderEarlyExit: true being set
@formbricks/web:build:     at /Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/export/index.js:481:23
@formbricks/web:build:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
@formbricks/web:build:     at async Promise.all (index 1)
@formbricks/web:build:     at async exportAppImpl (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/export/index.js:442:21)
@formbricks/web:build:     at async /Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/export/index.js:625:16
@formbricks/web:build:     at async Span.traceAsyncFn (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/trace/trace.js:157:20)
@formbricks/web:build:     at async /Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/build/index.js:1690:42
@formbricks/web:build:     at async Span.traceAsyncFn (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/trace/trace.js:157:20)
@formbricks/web:build:     at async /Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/build/index.js:1548:17
@formbricks/web:build:     at async Span.traceAsyncFn (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/trace/trace.js:157:20)
@formbricks/web:build:    Generating static pages (1/57)  [    ] ELIFECYCLE  Command failed with exit code 1.
@formbricks/web:build: ERROR: command finished with error: command (/Users/xstraza/OpinodoProjects/opinodo-surveys/apps/web) /opt/homebrew/bin/pnpm run build exited (1)

```

**Reply by amitamrutiya** [reply]
> i have a similar problem, using these versions:
> 
> ```
>     "next": "15.0.0-rc.0",
>     "optional": "^0.1.4",
>     "next-logger": "^4.0.0",
>     "otplib": "^12.0.1",
>     "papaparse": "^5.4.1",
>     "posthog-js": "^1.135.2",
>     "prismjs": "^1.29.0",
>     "qrcode": "^1.5.3",
>     "react": "19.0.0-rc-107a2f8c3e-20240617",
>     "react-dom": "19.0.0-rc-107a2f8c3e-20240617",
> ```
> 
> ```
>    Generating static pages (0/57)  [ ===]TypeError: Cannot read properties of null (reading 'useContext')
> @formbricks/web:build:     at exports.useContext (/Users/xstraza/OpinodoProjects/opinodo-surveys/apps/web/node_modules/react/cjs/react.production.js:489:33)
> @formbricks/web:build:     at h (/Users/xstraza/OpinodoProjects/opinodo-surveys/apps/web/.next/server/pages/_error.js:1:6021)
> @formbricks/web:build:     at renderWithHooks (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4007:18)
> @formbricks/web:build:     at renderElement (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4166:17)
> @formbricks/web:build:     at renderNodeDestructive (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4696:16)
> @formbricks/web:build:     at renderNode (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4921:14)
> @formbricks/web:build:     at renderChildrenArray (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4853:7)
> @formbricks/web:build:     at renderNodeDestructive (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4710:9)
> @formbricks/web:build:     at renderNode (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4921:14)
> @formbricks/web:build:     at renderElement (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4207:7)
> @formbricks/web:build: 
> @formbricks/web:build: Error occurred prerendering page "/500". Read more: https://nextjs.org/docs/messages/prerender-error
> @formbricks/web:build: 
> @formbricks/web:build: TypeError: Cannot read properties of null (reading 'useContext')
> @formbricks/web:build:     at exports.useContext (/Users/xstraza/OpinodoProjects/opinodo-surveys/apps/web/node_modules/react/cjs/react.production.js:489:33)
> @formbricks/web:build:     at h (/Users/xstraza/OpinodoProjects/opinodo-surveys/apps/web/.next/server/pages/_error.js:1:6021)
> @formbricks/web:build:     at renderWithHooks (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4007:18)
> @formbricks/web:build:     at renderElement (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4166:17)
> @formbricks/web:build:     at renderNodeDestructive (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4696:16)
> @formbricks/web:build:     at renderNode (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4921:14)
> @formbricks/web:build:     at renderChildrenArray (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4853:7)
> @formbricks/web:build:     at renderNodeDestructive (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4710:9)
> @formbricks/web:build:     at renderNode (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4921:14)
> @formbricks/web:build:     at renderElement (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/node_modules/react-dom/cjs/react-dom-server.browser.production.js:4207:7)
> 
> > Build error occurred
> Error: Export encountered an error on /500, exiting due to prerenderEarlyExit: true being set
> @formbricks/web:build:     at /Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/export/index.js:481:23
> @formbricks/web:build:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> @formbricks/web:build:     at async Promise.all (index 1)
> @formbricks/web:build:     at async exportAppImpl (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/export/index.js:442:21)
> @formbricks/web:build:     at async /Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/export/index.js:625:16
> @formbricks/web:build:     at async Span.traceAsyncFn (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/trace/trace.js:157:20)
> @formbricks/web:build:     at async /Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/build/index.js:1690:42
> @formbricks/web:build:     at async Span.traceAsyncFn (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/trace/trace.js:157:20)
> @formbricks/web:build:     at async /Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/build/index.js:1548:17
> @formbricks/web:build:     at async Span.traceAsyncFn (/Users/xstraza/OpinodoProjects/opinodo-surveys/node_modules/next/dist/trace/trace.js:157:20)
> @formbricks/web:build:    Generating static pages (1/57)  [    ] ELIFECYCLE  Command failed with exit code 1.
> @formbricks/web:build: ERROR: command finished with error: command (/Users/xstraza/OpinodoProjects/opinodo-surveys/apps/web) /opt/homebrew/bin/pnpm run build exited (1)
> ```

+1

**Comment by jrea** [comment]
`npm install react@18.3.1 react-dom@18.3.1 --save` did the trick for me. 

I assume in other cases, you need to match the react version if whatever `next` you're running with your installed version.


How I fixed this:

```
❯ npm ls react
...
 ├─┬ next@14.2.14
  │ ├─┬ react-dom@18.3.1
  │ │ └── react@18.3.1 deduped
  │ ├── react@18.3.1
  │ └─┬ styled-jsx@5.1.1
  │   └── react@18.3.1 deduped
...
```

found that in the monorepo, some deps had `18.2.0` (pinned in my repo previously), others had `18.3.0` (next 14 has `18.3.1`).

pinning `react` and `react-dom` to the nextjs react version (`18.3.1`) allowed the build to work.

**Comment by kyleatcolenso** [comment]
I'm running into the same problem, i bumped React from 18.3.1 down to 18.2 and it didn't work.

```
Error occurred prerendering page "/500". Read more: https://nextjs.org/docs/messages/prerender-error
...
TypeError: Cannot read properties of null (reading 'useContext')
...
150

[WARNING]: > Export encountered errors on following paths:
/_error: /404
/_error: /500
```

**Comment by ezengold** [comment]
@icbrahimc I answer the same problem already [here](https://github.com/vercel/next.js/discussions/43577#discussioncomment-11182679)

It is about using a fixed version of react in the `package.json` of a package in your monorepo. Put `^` before the version number.

**Comment by SwarnenduG07** [comment]
it worked for me when I went back to react 18.2 and react-dom 18.2

**Comment by steezeburger** [comment]
I had to delete my package-lock.json and npm modules and reinstall everything to build. I think you could run into an issue with ci/cd where cache is causing version mismatching between deps as well.
